rest/src/test/kotlin/systems/ajax/englishstudytelegrambot/NatsControllerTest.kt

```diff
@@ -52,189 +49,272 @@ class NatsControllerTest {
     private lateinit var libraryRepository: LibraryRepository
 
     @Autowired
-    private lateinit var wordRepository: WordRepository
+    private lateinit var userRepository: UserRepository
 
     @Autowired
-    private lateinit var adminService: AdminService
+    private lateinit var wordRepository: WordRepository
 
     @Autowired
-    private lateinit var mongoTemplate: MongoTemplate
-
-    @BeforeEach
-    fun clearDBs() {
-        mongoTemplate.remove<User>(Query())
-        mongoTemplate.remove<Library>(Query())
-        mongoTemplate.remove<Word>(Query())
-    }
-
-    private fun addLibrariesAndUsers(): List<Library> =
-        listOf(
-            libraryRepository.saveNewLibrary("testLibraryName1", "testTelegramUserId1"),
-            libraryRepository.saveNewLibrary("testLibraryName2", "testTelegramUserId2"),
-            libraryRepository.saveNewLibrary("testLibraryName3", "testTelegramUserId1")
-        )
+    private lateinit var adminService: AdminService
 
     private fun createEmptyAdditionalInfoAboutWord() = AdditionalInfoAboutWord("", "", "", "")
 
     @Test
     fun testGetAllUserNatsController() {
-
-        val libraries = addLibrariesAndUsers()
-
-        val message: Message = doRequest(GET_ALL_USERS_SUBJECT, GetAllUsersRequest.getDefaultInstance().toByteArray())
-
+        // GIVEN
+        val nanoTime = System.nanoTime().toString()
```

Comment:
NanoTime should be invoked every time you pass it to variable - Sooner or later your approach will confuse you ( IMHO)

Reply:
one nanoTime value  makes data inside method unique so i consider only one is enough